src/client/http-poller.ts

```diff
@@ -0,0 +1,84 @@
+import to from 'await-to-js';
+import HttpStatusCode from 'http-status-typed';
+
+export type HttpPoller = {
+  start: () => void;
+  stop: () => void;
+};
+
+export type HttpPollerProps = {
+  pollFunction: () => Promise<Response | undefined>;
+  shouldPoll: () => boolean;
+  onError: (returnedHttpStatus?: number) => { keepPolling: boolean };
+  pollInterval?: number;
```

Comment:
I like to include the time unit's name (milliseconds, seconds, hours...) to the variable name, if the unit is not obvious from variable's type (it's a pure number type). Something like `pollIntervalMillis` in this case, for example.

Reply:
Added "<variableName>InMs" to everywhere time is used. In js, all intervals and timeouts are always in milliseconds, so didn't put that in the name. But better to have it anyway.